lib/chef/knife/azurerm_server_create.rb

```diff
@@ -274,6 +274,53 @@ def ssh_override_winrm
           config[:ssh_password] = locate_config_value(:winrm_password)
         end
       end
+
+      def set_default_image_reference
+        if locate_config_value(:azure_image_os_type)
+          if (locate_config_value(:azure_image_reference_publisher) || locate_config_value(:azure_image_reference_offer) || locate_config_value(:azure_image_reference_sku))
+            # if azure_image_os_type is given and other image reference parameters are also given,
+            # raise error
+            ui.error("Please specify either --azure-image-os-type OR other image reference parameters i.e.
+              --azure-image-reference-publisher, --azure-image-reference-offer, --azure-image-reference-sku, --azure-image-reference-version.")
+            exit 1
```

Comment:
We should never exit, we should raise an exception that should be handled gracefully by the base command. That handler can then call `exit 1`. Also, we should name the method with a `!` so callers know this can terminate the application.

Note that if we're doing `exit 1` elsewhere, we can let this go in this PR, but we should make a new change to fix the error handling so it is consistent (it might actually be the case that `knife` itself handles errors in some way).